### PR TITLE
Fix search bar

### PR DESCRIPTION
**Pull request by JEFFKAY16**
### Fix Pagination on Search Functionality
Hey @AlexHooperDev - I'm making this PR with the following updates:

- Fixed the search functionality related to pagination and restoring the data when search(es) is/are deleted.
- I created a separate component for the table to keep things cleaner and less congested in the `Inventory.tsx` file

Please feel free to review this PR and leave some feedback or suggestions on what can b improved. I just saw an update on the `Dashboard Tasks ` called **Adding vertical columns to improve readability** - I'll go ahead and start working on it.

Thanks,
Jeff

### CODE REVIEW

File: src/components/InventoryTable.tsx [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+              Incoming Quantity
+            </TableCell>
+            {/* <TableCell sx={styles.tableHeaders}>
+              % of Incoming Stock Allocated To Flare Orders
+            </TableCell> */}
+          </TableRow>
+        </TableHead>
+        <TableBody>
+          {
+            currentRecords.map((product: any) => (
```
Comment (AlexHooperDev) [inline src/components/InventoryTable.tsx:69]: Can we assign the product to not be `any`? I think we have a type for it `ProductType` in our types file:?

File: src/components/InventoryTable.tsx [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+  "& .MuiSwitch-switchBase.Mui-checked + .MuiSwitch-track": {
+    backgroundColor: green[600],
+  },
+}))
+
+const label = { inputProps: { "aria-label": "Color switch demo" } }
+
+// PropTypes
+interface TableProps {
+  currentRecords: any
```
Comment (AlexHooperDev) [inline src/components/InventoryTable.tsx:34]: Can we make these into proper types?

File: src/components/InventoryTable.tsx [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+                          product.id,
+                          product.incomingQuantity
+                          )
+                        }
+                    >
+                      <EditIcon />
+                    </Button>
+                  </Box>
+                </TableCell>
+                {/* <TableCell sx={styles.green}>
```
Comment (AlexHooperDev) [inline src/components/InventoryTable.tsx:126]: Do we need this commented code? Probably best to delete for now and if we end up using the % allocated, we can go back through git history to copy and paste :)

File: src/components/SearchBar.tsx [hunk 4]
```diff
@@ -2,12 +2,12 @@ import React, { useEffect, useState } from "react"
...
 
 const SearchBar = ({ inputText, setInputText }: any) => {
-  const [debouncedInput, setDebouncedInput] = useState(inputText)
+  // const [debouncedInput, setDebouncedInput] = useState(inputText)
 
-  useEffect(() => {
-    const timer = setTimeout(() => setInputText(debouncedInput), 1000)
-    return () => clearTimeout(timer)
-  }, [debouncedInput])
+  // useEffect(() => {
```
Comment (AlexHooperDev) [inline src/components/SearchBar.tsx:7]: Why was this code removed? It was added to improve search performance :)

File: src/components/SearchBar.tsx [hunk 5]
```diff
@@ -16,10 +16,20 @@ const SearchBar = ({ inputText, setInputText }: any) => {
         placeholder="🔍 Search By Product Name"
         id="outlined-size-small"
         size="small"
-        onChange={(e) => setDebouncedInput(e.target.value)}
-        value={debouncedInput}
+        onChange={(e) => setInputText(e.target.value)}
+        value={inputText}
       />
     </Box>
+    // <Box>
```
Comment (AlexHooperDev) [inline src/components/SearchBar.tsx:23]: Commented code?

File: src/pages/Inventory.tsx [hunk 6]
```diff
@@ -1,129 +1,88 @@
...
-  // console.log(filteredData)
-  // console.log(inventoryData)
-
-  useEffect(() => {
-    // On first render, fetch inventory data and set it to state
     const fetchInventoryData = async (url: string) => {
       const fetchRequest = await fetch(url)
       const response = await fetchRequest.json()
-      setInventoryData(response.inventory)
+      setInventoryData((response.inventory).sort((firstProduct: any, secondProduct: any) => firstProduct.displayName.localeCompare(secondProduct.displayName)))
```
Comment (AlexHooperDev) [inline src/pages/Inventory.tsx:47]: Can we use a type other than any here please? :)

**Review by AlexHooperDev — CHANGES_REQUESTED** [review]
Looks good Jeff! Just a few changes / tweaks needed (see review comments)

File: src/components/InventoryTable.tsx (hunk 3 above)
Comment (JEFFKAY16) [reply src/components/InventoryTable.tsx:126]: This was the code responsible for assigning colors of `green`, `yellow`, or `red with danger/warning icon` to the percentage value. Yes sure, it's best we delete it, and if needed we can add it back. Thanks!

File: src/components/SearchBar.tsx (hunk 4 above)
Comment (JEFFKAY16) [reply src/components/SearchBar.tsx:7]: My bad, I couldn't figure out why this code was added plus I was trying to refactor the code. I have added it back, removed the commented out code and double checked to ensure everything still works.

File: src/pages/Inventory.tsx (hunk 6 above)
Comment (JEFFKAY16) [reply src/pages/Inventory.tsx:47]: My bad, I have added the appropriate type, many thanks for flagging these - I have just realized that by adding an `any` type throughout, we are defeating the purpose of `typescript` in our project. I'll keep an eye moving forward.

**Comment by JEFFKAY16** [comment]
Hey @AlexHooperDev - I have just made the required changes, many thanks for flagging out the parts where I was adding a type of `any` as I have just realized doing so defeats the purpose of `typescript` in our project. I'll keep an eye moving forward and if I can't tell what the type is, I'll reach out.

File: src/components/SearchBar.tsx [hunk 7]
```diff
@@ -2,12 +2,12 @@ import React, { useEffect, useState } from "react"
 import { Box, TextField } from "@mui/material"
 
 const SearchBar = ({ inputText, setInputText }: any) => {
```
Comment (AlexHooperDev) [inline src/components/SearchBar.tsx:4]: Can we create a prop type for these Props?

**Review by AlexHooperDev — APPROVED** [review]
Looks good Jeff! Just left a small comment to add a type to some props, but please merge otherwise :)

File: src/components/SearchBar.tsx (hunk 7 above)
Comment (JEFFKAY16) [reply src/components/SearchBar.tsx:4]: I have just incorporated this in the next PR. Thanks!

**Pull request merged**